server/src/main/java/com/cloud/network/NetworkModelImpl.java

```diff
@@ -2299,6 +2299,9 @@ public NicVO getPlaceholderNicForRouter(Network network, Long podId) {
                         ipv6 = _ipv6Dao.findByNetworkIdAndIp(network.getId(), nic.getIPv6Address());
                     }
                     //return nic only when its ip address belong to the pod range (for the Basic zone case)
+                    if (vlans.isEmpty()) {
+                        return nic;
+                    }
```

Comment:
I don't think this comment still describves the line below. Can you clarify or remove it?

Reply:
@DaanHoogland I've moved the comment